haystack/preview/components/websearch/utils.py

```diff
@@ -0,0 +1,39 @@
+from typing import List, Optional, Any
+
+from haystack.preview import Document
+
+
+def calculate_ranking_scores(list_items: List[Any], boost_first_factor: Optional[int] = None) -> List[float]:
+    """
+    Assigns scores to items in a list based on their rank position and ensures that the scores add up to 1.
```

Comment:
May be a dumb question, but why do we normalize the scores? They're not probabilities. Or did I miss something?

Reply:
Just to have some sense of how valuable a particular result link is. No other reason, otherwise we have some scale which is unknown and hard to understand.